src/tribler/core/utilities/async_group/async_group.py

```diff
@@ -3,38 +3,38 @@
 from contextlib import suppress
 from typing import Iterable, List, Set
 
+from tribler.core.utilities.async_group.exceptions import CancelledException
+
 
 class AsyncGroup:
```

Comment:
In the previous PR, `asyncio.create_task(...)` was replaced with `async_group.add_task(...)` whenever possible to avoid a situation when a task is garbage-collected and disappears before its execution ends.

But now, another case is possible when an async group itself is garbage-collected before all its tasks are finished. In that case, tasks managed by this async group can be garbage collected before they are finished.

Usually, it should not be a problem, as an async group is typically created as a field of another object with a long life span. So, someone holds a reference to AsyncGroup long enough to prevent its early garbage collection. But theoretically, some async groups can be garbage collected too early.

I'd like to see (maybe in some future PR) a `__del__` method of `AsyncGroup` that checks if this group manages some active futures. In that case, it should log an error message (it can't throw an exception, as [exceptions are suppressed inside `__del__`](https://docs.python.org/3/reference/datamodel.html#object.__del__))

Reply:
Done